rfcs/007 - Peer review infra changes.md

```diff
@@ -0,0 +1,125 @@
+# 007 - Peer review infrastructure changes RFC
+
+### Proposed
+
+2022-12-16
+
+### Accepted
+
+YYYY-MM-DD
+
+#### Approvers
+
+- David Ellis <david@iasql.com>
+- Yolanda Robla <yolanda@iasql.com>
+- Alejandro Guillen <alejandro@iasql.com>
+- Mohammad Pabandi <mohammad@iasql.com>
+
+### Implementation
+
+- [ ] Implemented: [One or more PRs](https://github.com/iasql/iasql-engine/some-pr-link-here) YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author(s)
+
+- Luis Fernando De Pombo <l@iasql.com>
+
+## Summary
+
+Eng teams move over from using cloud UIs to IaC because cloud infrastructure changes can lead to outages more often than business logic changes. As a result, it is considered a DevOps best practice to code review infrastructure changes using IaC tools and show the `preview` of the resulting change in pull requests to raise the quality bar and reduce the possibility of mistakes. Cloud infrastructure cannot be code reviewed with IaSQL using migration systems even if they support non-schema, or data, migrations (some ORMs like Prisma only support schema changes), and each migration is wrapped in an IaSQL transaction. Through our research and dogfood efforts, we have not found a migration system that fulfills the requirements needed to code review changes with IaSQL:
+- Allow data-only, `up` migrations
+- Allow plain SQL or have an introspective ORM
+- Free and ideally open source
+- Baseline/ignore certain migrations
+- Be resilient to `iasql_ugprade`
+
+Data migrations require being able to baseline migrations to avoid duplicate record violations on `iasql_upgrade`. Flyway comes closest to meeting all our requirements. However, it is not possible to baseline/ignore existing migrations on their free product.
```

Comment:
Same comment here wrt `iasql_upgrade`, just note that when IaSQL upgrades its schema it reconstructs the entire database so migration data will get dropped and therefore the migration system needs to be able to deal with duplicate record violations.